packages/medusa/src/api/routes/store/carts/create-cart.ts

```diff
@@ -125,21 +124,18 @@ export default async (req, res) => {
     }
   }
 
-  if (featureFlagRouter.isFeatureEnabled(PublishableAPIKeysFeatureFlag.key)) {
-    if (
-      !toCreate.sales_channel_id &&
-      req.publishableApiKeyScopes?.sales_channel_id.length
-    ) {
-      if (req.publishableApiKeyScopes.sales_channel_id.length > 1) {
-        throw new MedusaError(
-          MedusaError.Types.UNEXPECTED_STATE,
-          "The PublishableApiKey provided in the request header has multiple associated sales channels."
-        )
-      }
-
-      toCreate.sales_channel_id =
-        req.publishableApiKeyScopes.sales_channel_id[0]
+  if (
+    !toCreate.sales_channel_id &&
+    req.publishableApiKeyScopes?.sales_channel_id.length
```

Comment:
suggestion: `sales_channel_id` looks like it is a value but in fact it is a collection, should it be `sales_channel_ids`?